src/main/java/com/gitblit/wicket/pages/EditRepositoryPage.java

```diff
@@ -707,6 +714,25 @@ public void onClick() {
 		}
 		form.add(delete.setVisible(canDelete));
 
+		boolean canRunGC = canDelete;

```

Comment:
Differentiate also if gc is running to either:
* disable button 
* or show that gc is running